library/x509_crt.c

```diff
@@ -3040,12 +3040,53 @@ static int x509_crt_check_san( const mbedtls_x509_buf *name,
     if( san_type == MBEDTLS_X509_SAN_DNS_NAME )
         return( x509_crt_check_cn( name, cn, cn_len ) );
 
+    /* iPAddress */
+    if( san_type == MBEDTLS_X509_SAN_IP_ADDRESS )
+        return ( ( *san_ip = 1 ) );
+
     /* (We may handle other types here later.) */
 
     /* Unrecognized type */
     return( -1 );
 }
 
+/* mbedtls will likely want some abstraction here in order to support systems
+ * without inet_pton() or without IPv6 or with different includes for structs */
```

Comment:
Well, yes, we can't just break the build on platforms without `inet_pton`. This should probably be a platform abstraction (`MBEDTLS_PLATFORM_INET_PTON_MACRO`, defaulting to `inet_pton`). The platform macro isn't enough though: how do we deal with the requirement to include different headers on different platforms?

Data point: Mbed OS does have `inet_pton`. (I haven't checked whether it's a complete implementation, but the function is present in the source code, it's part of LWIP.) So it's not just a “big system” thing.

Reply:
Since it's not just a “big system” thing, it's good to reuse the platform's function if it exists, to save code size.

Reply:
> The platform macro isn't enough though: how do we deal with the requirement to include different headers on different platforms?

Would it work to have another macro for the header name as well?

Reply:
`#include MACRO_NAME` is not standard C, though it's widely supported, and we're already using
```
#if defined(MACRO_NAME)
#include MACRO_NAME
#endif
```
in our code base (for `MBEDTLS_USER_CONFIG_FILE`) with no complaints that I've seen.

`#include MACRO_NAME` doesn't address systems that need user code to include multiple headers. OpenBSD used to do this a lot, but I think they've come around now, though [the `inet_pton` man page](https://man.openbsd.org/inet_pton) still lists two headers (maybe only one is enough in practice though).

Reply:
As requested below, I have modified the PR to fall back to the local `inet_pton()` implementation that I have provided if `__has_include()` is not supported by the compiler.  That is a coarse heuristic, which could be modified later to use the platform implementation, where desired.  FYI: the code size of the local `inet_pton()` implementation is < 500 bytes  more than using the platform `inet_pton()` (in a simple `gcc -Os` test on my system)
:rocket: